pkg/inventory/inventorycm.go

```diff
@@ -119,6 +125,69 @@ func (icm *ConfigMap) GetObject() (*unstructured.Unstructured, error) {
 	return invCopy, nil
 }
 
+// Apply is an Storage interface function implemented to apply the inventory
+// object. StatusPolicy is not needed since ConfigMaps do not have a status subresource.
+func (icm *ConfigMap) Apply(dc dynamic.Interface, mapper meta.RESTMapper, _ StatusPolicy) error {
+	invInfo, namespacedClient, err := icm.getNamespacedClient(dc, mapper)
+	if err != nil {
+		return err
+	}
+
+	// Get cluster object, if exsists.
+	clusterObj, err := namespacedClient.Get(context.TODO(), invInfo.GetName(), metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	// Create cluster inventory object, if it does not exist on cluster.
+	if clusterObj == nil {
+		klog.V(4).Infof("creating inventory object: %s/%s", invInfo.GetNamespace(), invInfo.GetName())
+		_, err = namespacedClient.Create(context.TODO(), invInfo, metav1.CreateOptions{})
+		return err
+	}
+
+	// Update the cluster inventory object instead.
+	klog.V(4).Infof("updating inventory object: %s/%s", invInfo.GetNamespace(), invInfo.GetName())
+	_, err = namespacedClient.Update(context.TODO(), invInfo, metav1.UpdateOptions{})
+	return err
+}
+
+// ApplyWithPrune is a Storage interface function implemented to apply the inventory object with a list of objects
+// to be pruned. StatusPolicy is not needed since ConfigMaps do not have a status subresource.
```

Comment:
This PR changes only the handling of the `Status` sub-resource. For ConfigMaps, since there is no status sub-resource, we can ignore the `statusPolicy` argument since we do not need to also update the `Status` sub-resource. The logic to handle storing status within the ConfigMap `data` using `StatusPolicy` is untouched.

Reply:
The point of adding the StatusPolicy in the first place was to be able to reduce the size of the object.

The fact that we reduce the size by not applying the status subresource is an implementation detail and should be implimentation-specific.

Therefore, if the ConfigMap inventory is not respecting the StatusPolicy, that's a bug. We should fix that, either in this PR or a follow up.

Reply:
I believe the current logic still respects StatusPolicy for ConfigMaps. The status is stored within `icm.objStatus` (eg: https://github.com/kubernetes-sigs/cli-utils/blob/302250d9e8d7fa8304cfd879ec69eae1e1ab34f0/pkg/inventory/inventorycm.go#L100). There are other code brancehs for StatusPolicy in inventory-client.go (eg: https://github.com/kubernetes-sigs/cli-utils/blob/302250d9e8d7fa8304cfd879ec69eae1e1ab34f0/pkg/inventory/inventory-client.go#L212) where we determine whether to inject the status into the ConfigMap object or not. This ultimately decides if the status is stored within the ConfigMap data or not.

The handling for StatusPolicy in `Apply` and `ApplyWithPrune` is purely for whether we need to call `UpdateStatus` for updating the status subresource in the inventory object on cluster.

NB: tested with locally built `kapply` binary to confirm that StatusPolicy for ConfigMaps is respected.

Reply:
Thanks for clarifying.